hal/api/Serial.h

```diff
@@ -74,6 +74,13 @@ class Serial : public SerialBase, public Stream {
     PlatformMutex _mutex;
 };
 
+/** Get the stdio Serial object, which is lazy instantiated. There's only 
+ *  one stdio object that should be used within an application.
+ *
+ *  @returns stdio object 
+ */
+Serial& get_stdio_serial();
```

Comment:
Small question, should this be a reference? This could confuse the ownership of the Serial object, and I'm not sure what the benefit is over `Serial*`.

With `Serial&` it's easy to accidentally do this:
``` cpp
void maximize_baud() {
    Serial s = get_stdio_serial();
    s.baud(96000000);
    // s's destructor is called, possibly deinitializing serial
}
```